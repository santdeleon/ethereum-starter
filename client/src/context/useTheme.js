import React, { createContext, useContext } from 'react';
import {
  ThemeProvider as StyledComponentsThemeProvider,
  createGlobalStyle,
} from 'styled-components';
import { oneOfType, array, object } from 'prop-types';

import { useLocalStorage } from './useLocalStorage';

const propTypes = {
  children: oneOfType([array, object]),
};

const LightTheme = {
  color: '#000',
  backgroundColor: '#fff',
};

const DarkTheme = {
  color: '#fff',
  backgroundColor: '#222',
};

const GlobalStyles = createGlobalStyle`
  body {
    backgroundColor: ${({ theme }) => theme.backgroundColor};
    color: ${({ theme }) => theme.color};
  }
`;

const ThemeContext = createContext();

const ThemeProvider = ({ children }) => {
  const prefersOSDarkTheme =
    window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)');

  const [theme, setTheme] = useLocalStorage(
    'theme',
    prefersOSDarkTheme ? 'light' : 'dark',
  );

  const toggleTheme = () => setTheme(theme === 'light' ? 'dark' : 'light');

  return (
    <ThemeContext.Provider value={{ theme, toggleTheme }}>
      <StyledComponentsThemeProvider theme={theme === 'light' ? LightTheme : DarkTheme}>
        {children}
      </StyledComponentsThemeProvider>
    </ThemeContext.Provider>
  );
};

const useTheme = () => {
  const context = useContext(ThemeContext);
  if (!context) throw new Error('You must useTheme within a <ThemeProvider />');
  return context;
};

ThemeProvider.propTypes = propTypes;
export { ThemeProvider, GlobalStyles, LightTheme, DarkTheme, useTheme };
