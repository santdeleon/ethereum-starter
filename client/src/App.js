import React from 'react';

import './App.css';

// import SimpleStorageContract from './contracts/SimpleStorage.json';

import { GlobalStyles } from './context/ThemeContext';

const App = () => {
  return (
    <div className="App">
      <GlobalStyles />
      <h1>Hello, Ethereum World!</h1>
    </div>
  );
};

export default App;
