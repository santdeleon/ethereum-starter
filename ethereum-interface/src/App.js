import React, { useEffect, useState, useCallback } from 'react';
import { Web3Provider } from '@ethersproject/providers';
// import { useWeb3React } from '@web3-react/core';

import { formatFixed } from '@ethersproject/bignumber';

import './App.css';

import SimpleStorageContract from './contracts/SimpleStorage.json';

import { GlobalStyles } from './context/ThemeContext';

import { useEagerConnect } from './hooks/useEagerConnect';
import { getContract } from './utils/getContract';

const App = () => {
  const [inputVal, setInputVal] = useState('');
  const [numberFromStorage, setNumberFromStorage] = useState(null);

  useEagerConnect();
  // const { active, account, chainId, library, error } = useWeb3React();

  const contract = getContract(
    SimpleStorageContract.networks[5777].address,
    SimpleStorageContract.abi,
    new Web3Provider(window.ethereum).getSigner(),
  );

  const storeNumberToBlockchain = async () => {
    const tx = await contract.set(parseInt(inputVal));
    await tx.wait(1);
    setInputVal('');
    alert(`Nice: 🦄🦄🦄🦄🦄🌈🌈🌈🌈🌈🌈`);
  };

  const getNumberFromBlockchain = useCallback(async () => {
    const tx = await contract.get();
    setNumberFromStorage(parseInt(formatFixed(tx._hex)));
  }, [contract]);

  useEffect(() => {
    getNumberFromBlockchain();
  }, [getNumberFromBlockchain]);

  return (
    <div className="App">
      <GlobalStyles />
      <h1>Hello, Ethereum World!</h1>
      <h3 style={{ color: 'gray', fontWeight: '100' }}>
        Test out your Simple Storage Contract by storing a number to the blockchain.
      </h3>

      <div>
        <input
          type="text"
          value={inputVal}
          onChange={(e) => setInputVal(e.target.value)}
          style={{
            borderRadius: '8px',
            border: '1px solid lightgray',
            marginBottom: '5px',
          }}
        />
      </div>
      <button
        onClick={storeNumberToBlockchain}
        style={{
          border: '1px solid #bada55',
          backgroundColor: '#bada55',
          padding: '5px 10px',
          borderRadius: '8px',
          marginTop: '5px',
        }}
      >
        <span role="img" aria-label="Rainbow Emoji">
          🌈
        </span>{' '}
        Submit
      </button>

      <p style={{ marginBottom: '20px' }}>
        {numberFromStorage ? (
          <>
            Your Number!{' '}
            <span role="img" aria-label="Party Popper Emoji">
              🎉🎉🎉🎉
            </span>
          </>
        ) : (
          <>
            Plzzz enter a number{' '}
            <span role="img" aria-label="Two Hearts Emoji">
              💕🥺
            </span>
          </>
        )}
      </p>
      {numberFromStorage && (
        <span
          style={{
            border: '1px solid pink',
            padding: '5px 10px',
            borderRadius: '8px',
            backgroundColor: 'pink',
          }}
        >
          {numberFromStorage}
        </span>
      )}
    </div>
  );
};

export default App;
