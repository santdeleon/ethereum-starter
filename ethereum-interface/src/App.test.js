import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';

import App from './App';

import { ThemeProvider } from './context/ThemeContext';

describe('the App component', () => {
  it('renders content a welcome message', async () => {
    const { container } = render(
      <ThemeProvider>
        <App />
      </ThemeProvider>,
    );

    expect(
      screen.getByRole('heading', { name: 'Hello, Ethereum World!' }),
    ).toBeInTheDocument();

    await waitFor(() => expect(container).toMatchSnapshot());
  });
});
