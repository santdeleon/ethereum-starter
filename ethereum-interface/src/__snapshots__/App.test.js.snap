// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the App component renders content a welcome message 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Hello, Ethereum World!
    </h1>
  </div>
</div>
`;
